driver/src/main/com/mongodb/ListCollectionsFluentImpl.java

```diff
@@ -83,7 +83,7 @@ public T first() {
     }
 
     @Override
-    public <U> MongoIterable<U> map(final Function<T, U> mapper) {
+    public <U> MappingIterable<T, U> map(final Function<T, U> mapper) {
```

Comment:
I guess this doesn't hurt, but is it necessary, as both this class and MappingIterable are not-public?  Maybe there's a test that requires it that I haven't seen yet.